pymongo/monitoring.py

```diff
@@ -187,12 +187,12 @@ def connection_checked_in(self, event):
 
 import datetime
 from collections import abc, namedtuple
-from typing import TYPE_CHECKING, Any, Dict, Optional
+from typing import TYPE_CHECKING, Any, Dict, Mapping, Optional
 
 from bson.objectid import ObjectId
 from pymongo.hello import Hello, HelloCompat
 from pymongo.helpers import _handle_exception
-from pymongo.typings import _Address, _DocumentOut
```

Comment:
I believe we need to keep using _DocumentOut in this file since `MutableMapping|RawBSONDocument` is the more specific type and we want to return the most specific types we can.

Reply:
Makes sense, I added that back in. 

I had initially removed it because I thought the `_DocumentOut` types were somehow associated with the `_DocumentIn` type. like if a user passed us a Mapping, we'd then give back a Mapping. But if we create a new object and give that back (which i'm guessing is what we do), then it would make sense to keep `_DocumentOut`